addons/account/models/account.py

```diff
@@ -1375,29 +1375,40 @@ def _check_repartition_lines(self, lines):
         base_line = lines.filtered(lambda x: x.repartition_type == 'base')
         if len(base_line) != 1:
             raise ValidationError(_("Invoice and credit note repartition should each contain exactly one line for the base."))
+        tax_lines = lines - base_line
+        if not tax_lines:
+            if base_line.factor_percent == 100:
+                values = {
+                    'repartition_type': 'tax',
+                    'factor_percent': 100.0,
+                    'company_id': base_line.company_id.id,
+                    'country_id': base_line.country_id.id,
+                }
+                self.env['account.tax.repartition.line'].create([
+                    {**values, 'invoice_tax_id': self.id},
+                    {**values, 'refund_tax_id': self.id},
+                ])
+            else:
+                raise ValidationError(_("Invoice and credit note repartition should have at least one tax repartition line."))
 
     @api.constrains('invoice_repartition_line_ids', 'refund_repartition_line_ids')
     def _validate_repartition_lines(self):
         for record in self:
-            invoice_repartition_line_ids = record.invoice_repartition_line_ids.sorted()
-            refund_repartition_line_ids = record.refund_repartition_line_ids.sorted()
```

Comment:
It's 3 years ago but that was added for performance reason.
Do we know if it still has any effect now? (especially as you are still sorting those later on)

Reply:
It doesn't have any impact if you look at the code below?